scico/test/optimize/test_admm.py

```diff
@@ -145,6 +148,49 @@ def test_admm_generic(self):
         x = admm_.solve()
         assert (snp.linalg.norm(self.grdA(x) - self.grdb) / snp.linalg.norm(self.grdb)) < 1e-3
 
+    def test_admm_saveload(self):
```

Comment:
Does this test represent the intended use case for state saving? It doesn't look like it to me, but I'm not sure. Could we instead do something like (1) create optimizer A (2) run A for e.g. 2 steps, save the state (3) create optimizer B, load the state from A, run 2 steps. (4) create C, run for 4 steps. (5) compare the result of B and C.

Reply:
It's not exactly replicating the intended use case, which is to be able to restart from a checkpoint. The approach you describe is quite similar to the current test - see details in my other response.